### DISCUSSION TITLE
다양한 컨텍스트(화살표 함수, 일반 함수, 메소드, 이벤트 핸들러)에서 this의 값이 어떻게 결정되는지 구체적인 예를 들어 설명해 주실 수 있나요?

### DISCUSSION THREAD

**Discussion by lledellebell** [diving-javascript/you-dont-know-js-yet#25] [실행 컨텍스트(Execution Context)]
- #18

**Comment by wogus216** [comment]
this는 실행 컨텍스트가 생성될 때 결정된다.
this 값은 주로 다음 네 가지 방법

일반함수: 함수가 일반적으로 호출되었기 때문에 this는 글로벌 객체를 가리킵니다. 참고, . 엄격 모드('use strict')에서는 this가 undefined로 설정됩니다.
메소드: 객체의 메소드로서 함수가 호출될 때, this는 그 메소드를 호출한 객체를 가리킵니다.
화살표 함수: this는 항상 상위 스코프의 this를 가리킵니다. 화살표 함수 자체는 this에 대한 자체 바인딩을 가지지 않습니다.
브라우저에서 DOM 요소에 이벤트 핸들러로 함수를 지정할 때, this는 이벤트가 발생한 요소를 가리킵니다.
this 바인딩의 특수 케이스
화살표 함수: 화살표 함수는 자신의 this를 가지지 않습니다. 대신, 화살표 함수는 자신을 포함하고 있는 외부 함수의 this를 "상속"받습니다. 이것은 렉시컬 스코핑(lexical scoping)이라고 하며, 화살표 함수가 정의된 위치에 따라 this가 결정됩니다.
bind/call/apply 메소드: 이 메소드들은 함수의 this 값을 명시적으로 설정할 수 있습니다. bind는 새로운 함수를 반환하며, call과 apply는 즉시 함수를 호출합니다. 이 메소드들을 사용하면 개발자는 함수 호출 시 this의 값이 무엇인지 정확히 지정할 수 있습니다.
